### PR TITLE
[libc] Add proxy headers for fenv types.

### PR DESCRIPTION
**Pull request by ghost**
Hello, here is my pr for #88187.

I have ran he build on overlay mode and full build mode and also ran the tests by doing `ninja check-libc`.

I have also ran the bazel build.

everything looks good.

please let me know of anything I need to change. thanks

### CODE REVIEW

File: libc/src/__support/FPUtil/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,8 @@ add_header_library(
   HDRS
     FEnvImpl.h
   DEPENDS
-    libc.include.fenv
+    libc.hdr.fenv_t
+    libc.hdr.fexcept_t
```
Comment (lntue) [inline libc/src/__support/FPUtil/CMakeLists.txt:7]: Remove `libc.hdr.fexcept_t` dependency, `FEnvImpl.h` does not use `fexcept_t`.

File: libc/src/fenv/CMakeLists.txt [hunk 2]
```diff
@@ -5,7 +5,8 @@ add_entrypoint_object(
   HDRS
     fegetround.h
   DEPENDS
-    libc.include.fenv
+    libc.hdr.fenv_t
+    libc.hdr.fexcept_t
```
Comment (lntue) [inline libc/src/fenv/CMakeLists.txt:9]: Remove both of these dependency, they are not directly used in `fegetround.(h|cpp)`.

File: libc/src/fenv/fegetenv.h [hunk 3]
```diff
@@ -9,7 +9,7 @@
 #ifndef LLVM_LIBC_SRC_FENV_FEGETENV_H
 #define LLVM_LIBC_SRC_FENV_FEGETENV_H
 
-#include <fenv.h>
+#include "hdr/fenv_t.h"
```
Comment (lntue) [inline libc/src/fenv/fegetenv.h:12]: Add `libc.hdr.fenv_t` to its direct dependency.

File: libc/src/fenv/fegetexceptflag.cpp [hunk 4]
```diff
@@ -7,11 +7,10 @@
 //===----------------------------------------------------------------------===//
 
 #include "src/fenv/fegetexceptflag.h"
+#include "hdr/fexcept_t.h"
```
Comment (lntue) [inline libc/src/fenv/fegetexceptflag.cpp:10]: Add libc.hdr.fenv_t to its direct dependency.

File: libc/src/fenv/feholdexcept.cpp [hunk 5]
```diff
@@ -7,9 +7,9 @@
 //===----------------------------------------------------------------------===//
 
 #include "src/fenv/feholdexcept.h"
+#include "hdr/fenv_t.h"
```
Comment (lntue) [inline libc/src/fenv/feholdexcept.cpp:10]: Add `libc.hdr.fenv_t` to its direct dependency.

File: libc/src/fenv/feholdexcept.h [hunk 6]
```diff
@@ -9,7 +9,7 @@
 #ifndef LLVM_LIBC_SRC_FENV_FEHOLDEXCEPT_H
 #define LLVM_LIBC_SRC_FENV_FEHOLDEXCEPT_H
 
-#include <fenv.h>
+#include "hdr/fenv_t.h"
```
Comment (lntue) [inline libc/src/fenv/feholdexcept.h:12]: Add `libc.hdr.fenv_t` to its direct dependency.

File: libc/src/fenv/fesetenv.h [hunk 7]
```diff
@@ -9,7 +9,7 @@
 #ifndef LLVM_LIBC_SRC_FENV_FESETENV_H
 #define LLVM_LIBC_SRC_FENV_FESETENV_H
 
-#include <fenv.h>
+#include "hdr/fenv_t.h"
```
Comment (lntue) [inline libc/src/fenv/fesetenv.h:12]: Add `libc.hdr.fenv_t` to its direct dependency.

File: libc/src/fenv/fesetexceptflag.cpp [hunk 8]
```diff
@@ -7,9 +7,9 @@
 //===----------------------------------------------------------------------===//
 
 #include "src/fenv/fesetexceptflag.h"
+#include "hdr/fexcept_t.h"
```
Comment (lntue) [inline libc/src/fenv/fesetexceptflag.cpp:10]: Add `libc.hdr.fexcept_t` to its direct dependency.

File: libc/src/fenv/feupdateenv.h [hunk 9]
```diff
@@ -9,7 +9,7 @@
 #ifndef LLVM_LIBC_SRC_FENV_FEUPDATEENV_H
 #define LLVM_LIBC_SRC_FENV_FEUPDATEENV_H
 
-#include <fenv.h>
+#include "hdr/fenv_t.h"
```
Comment (lntue) [inline libc/src/fenv/feupdateenv.h:12]: Add `libc.hdr.fenv_t` to its direct dependency.

File: libc/test/UnitTest/FPExceptMatcher.cpp [hunk 10]
```diff
@@ -8,7 +8,7 @@
 
 #include "FPExceptMatcher.h"
 
-#include <fenv.h>
+#include "hdr/fenv_t.h"
```
Comment (lntue) [inline libc/test/UnitTest/FPExceptMatcher.cpp:11]: This file also call `fegetenv` and `fesetenv` from `fenv.h`.  Can you replace them with `fputil::get_env` and `fputil::set_env` respectively?  Also include `src/FPUtil/FEnvImpl.h` and update its dependency accordingly.

**Review by lntue — APPROVED** [review]

File: libc/hdr/CMakeLists.txt [hunk 11]
```diff
@@ -41,6 +41,24 @@ add_proxy_header_library(
     libc.include.fenv
 )
 
+add_proxy_header_library(
+  fenv_t
```
Comment (nickdesaulniers) [inline libc/hdr/CMakeLists.txt:45]: types should go in libc/hdrs/types/

**Review by nickdesaulniers — CHANGES_REQUESTED** [review]

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel [hunk 12]
```diff
@@ -119,6 +119,15 @@ libc_support_library(
 )
 
 libc_support_library(
+    name = "hdr_fenv_t",
```
Comment (michaelrj-google) [inline utils/bazel/llvm-project-overlay/libc/BUILD.bazel:122]: once the source files are moved to the `/types/` folder, these targets should be moved down in this file to match.

File: libc/hdr/CMakeLists.txt (hunk 11 above)
Comment (Jobhdez) [reply libc/hdr/CMakeLists.txt:45]: ok thanks

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel (hunk 12 above)
Comment (Jobhdez) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel:122]: what do you mean by "should be moved down in this file to match"?  thanks by the way

File: libc/hdr/CMakeLists.txt (hunk 11 above)
Comment (Jobhdez) [reply libc/hdr/CMakeLists.txt:45]: and do I need to change the corresponding `libc.hdr.fenv_t` to `libc.hdr.types.fenv_t` in the CMakeList.txt?

Comment (nickdesaulniers) [reply libc/hdr/CMakeLists.txt:45]: yeah

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel (hunk 12 above)
Comment (nickdesaulniers) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel:122]: See  L140 `############################ Type Proxy Header Files ###########################`

Comment (Jobhdez) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel:122]: oh ok. thanks

File: libc/hdr/CMakeLists.txt (hunk 11 above)
Comment (Jobhdez) [reply libc/hdr/CMakeLists.txt:45]: thanks

Comment (Jobhdez) [reply libc/hdr/CMakeLists.txt]: Ok I have addressed your comments

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel (hunk 12 above)
Comment (Jobhdez) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel]: I have addressed your comments

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel [hunk 13]
```diff
@@ -154,6 +145,16 @@ libc_support_library(
...
 
+libc_support_library(
+    name = "types_fenv_t",
+    hdrs = ["hdr/types/fenv_t.h"],
+)
+
+libc_support_library(
+    name = "types_fexcept_t",
+    hdrs = ["hdr/types/fexcept_t.h"],
+)
```
Comment (nickdesaulniers) [inline utils/bazel/llvm-project-overlay/libc/BUILD.bazel:156]: mind trying to keep these sorted with regard to `sigset_t` and the rest? In that case, these should go above the sigset_t (L133-L136)

**Review by nickdesaulniers — COMMENTED** [review]
looking good, almost there!

File: utils/bazel/llvm-project-overlay/libc/BUILD.bazel (hunk 13 above)
Comment (Jobhdez) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel:156]: Will do.

Comment (Jobhdez) [reply utils/bazel/llvm-project-overlay/libc/BUILD.bazel]: Ok - I have just gotten done

**Review by nickdesaulniers — APPROVED** [review]
looks great! nice job, thank you!

need us to merge?  if so will do so next week (prefer not to merge Friday afternoon when it's time for 🍻

**Comment by Jobhdez** [comment]
> looks great! nice job, thank you!
> 
> need us to merge? if so will do so next week (prefer not to merge Friday afternoon when it's time for 🍻

Sounds good! Enjoy your 🍻

**Review by michaelrj-google — APPROVED** [review]

**Pull request merged**